From 6b23efbf03f39b629e1dc0892eccfb36552e186a Mon Sep 17 00:00:00 2001
From: Devajith Valaparambil Sreeramaswamy
 <devajith.valaparambil.sreeramaswamy@cern.ch>
Date: Thu, 31 Oct 2024 09:32:21 +0100
Subject: [PATCH 48/52] Revert "Find an easier way of accessing CodeGenModule
 internals."

This reverts commit 65c11ae6156f45136ffed18836dc2a61021e2950.
---
 interpreter/llvm-project/clang/lib/CodeGen/CodeGenModule.h   | 2 --
 interpreter/llvm-project/clang/lib/CodeGen/ModuleBuilder.cpp | 2 +-
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/interpreter/llvm-project/clang/lib/CodeGen/CodeGenModule.h b/interpreter/llvm-project/clang/lib/CodeGen/CodeGenModule.h
index fc256eb8b6..d9ece4d98e 100644
--- a/interpreter/llvm-project/clang/lib/CodeGen/CodeGenModule.h
+++ b/interpreter/llvm-project/clang/lib/CodeGen/CodeGenModule.h
@@ -59,7 +59,6 @@ class FileSystem;
 namespace clang {
 class ASTContext;
 class AtomicType;
-class CodeGeneratorImpl; // hack needed by cling
 class FunctionDecl;
 class IdentifierInfo;
 class ObjCImplementationDecl;
@@ -1770,7 +1769,6 @@ private:
 
   llvm::Metadata *CreateMetadataIdentifierImpl(QualType T, MetadataTypeMap &Map,
                                                StringRef Suffix);
-  friend class clang::CodeGeneratorImpl; // hack needed by cling
 };
 
 }  // end namespace CodeGen
diff --git a/interpreter/llvm-project/clang/lib/CodeGen/ModuleBuilder.cpp b/interpreter/llvm-project/clang/lib/CodeGen/ModuleBuilder.cpp
index dbb6fd10d4..3594f4c66e 100644
--- a/interpreter/llvm-project/clang/lib/CodeGen/ModuleBuilder.cpp
+++ b/interpreter/llvm-project/clang/lib/CodeGen/ModuleBuilder.cpp
@@ -29,7 +29,7 @@
 using namespace clang;
 using namespace CodeGen;
 
-namespace clang {
+namespace {
   class CodeGeneratorImpl : public CodeGenerator {
     DiagnosticsEngine &Diags;
     ASTContext *Ctx;
-- 
2.43.5

